integr8ly/integreatly-operator
# MGDAPI-6011 - refactor RHOAM to work with PKO Grafana

# Issue link
Jira: https://issues.redhat.com/browse/MGDAPI-6011

## Latest updates:
- Validation done on OSD cluster, using image from MR https://gitlab.cee.redhat.com/service/managed-tenants-bundles/-/merge_requests/1102

```
CONFIG_IMAGE=quay.io/vmogilev_rhmi/managed-api-service-config:MGDAPI-5991Int3 make cluster/prepare/local
LOCAL=false USE_CLUSTER_STORAGE=true INSTALLATION_TYPE=managed-api make code/run
```
- Pipeline (with latest image): https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/job/Playground/job/vmogilev/job/test-pr-upgrade-6011/

**NOTES**
- PR will be merged Only after Grafana feature branch https://gitlab.cee.redhat.com/vmogilev/managed-tenants-bundles/-/tree/mgdapi-5991-grafana-int

# What
Refactor Rhoam to work with PKO Grafana

# Verification steps

1. Prepare OSD cluster and run Rhoam prepare local command as below (example with my  image `CONFIG_IMAGE=quay.io/vmogilev_rhmi/managed-api-service-config:MGDAPI-5991Int`) ,   
or create your own image with Grafana as noted in MR: https://gitlab.cee.redhat.com/vmogilev/managed-tenants-bundles/-/merge_requests/4, and run command similar to below:

```
$ CONFIG_IMAGE=quay.io/vmogilev_rhmi/managed-api-service-config:MGDAPI-5991Int make cluster/prepare/local
```
3. Rhoam Install
```
$ LOCAL=false USE_CLUSTER_STORAGE=true INSTALLATION_TYPE=managed-api make code/run
```

- **Check Grafana**
```
$ date
Thu Sep 28 15:59:35 IDT 2023
[vmogilev@vmogilev integreatly-operator] (MGDAPI-6011)$ oc get all -n redhat-rhoam-customer-monitoring
NAME                                     READY   STATUS    RESTARTS      AGE
pod/grafana-deployment-f856d9df6-89xp8   2/2     Running   6 (19m ago)   22m

NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
service/grafana-service   ClusterIP   172.30.34.87   <none>        3000/TCP,9091/TCP   22m

NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/grafana-deployment   1/1     1            1           22m

NAME                                           DESIRED   CURRENT   READY   AGE
replicaset.apps/grafana-deployment-f856d9df6   1         1         1       22m

NAME                                     HOST/PORT                                                                        PATH   SERVICES          PORT            TERMINATION   WILDCARD
route.route.openshift.io/grafana-route   grafana-route-redhat-rhoam-customer-monitoring.apps.vmo03.edy6.s1.devshift.org   /      grafana-service   grafana-proxy   reencrypt     None

```

![Screenshot from 2023-10-08 12-54-38](https://github.com/integr8ly/integreatly-operator/assets/56625811/887cda0a-fe49-4aea-a655-21c6dca392a3)

- Login as admin to Grafana and check that Datasourse is set to Prometheus
![Screenshot from 2023-10-08 11-54-46](https://github.com/integr8ly/integreatly-operator/assets/56625811/4f293bc8-9e87-4e12-a502-57c9f131ed01)

- Compare current Grafana with Grafana on `long-lived` cluster: https://grafana-route-redhat-rhoami-customer-monitoring-operator.apps.eng-long-lived.jpny.s1.devshift.org/d/66ab72e0d012aacf34f907be9d81cd9e/rate-limiting?orgId=1&refresh=1m

- **Check Rhoam**
```
$ oc get rhmi rhoam -n redhat-rhoam-operator -o json | jq -r '.status.stage'
complete

```
## Pipelines to check E2E
### Install:
https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/job/Playground/job/vmogilev/job/managed-api-install-master-6011/11/
**Notes**:  The following tests have been corrected based on the analysis of results:  **C04 and A27 + A28**. 

- Upgrade pipeline is in work:
